### PR TITLE
[PAYSHIP-2922] new controller for Apple Pay

### PR DESCRIPTION
**Pull request by L3RAZ**

### CODE REVIEW

File: controllers/front/applepay.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            }
+
+            $action = $bodyValues['action'];
+
+            $this->commandBus = $this->module->getService('ps_checkout.bus.command');
+
+            if ($action === 'getPaymentRequest') {
+                $this->getPaymentRequest($bodyValues);
+            } else {
+                $this->exitWithExceptionMessage(new Exception('Invalid request', 400));
```
Comment (seiwan) [inline controllers/front/applepay.php:61]: Better to use idiomatic code, no ?
```suggestion
                $exception = new Exception('Invalid request', 400);
                $this->exitWithExceptionMessage($exception);
```

File: controllers/front/applepay.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            }
+        } catch (Exception $exception) {
+            $this->exitWithExceptionMessage($exception);
+        }
+    }
+
+    private function getPaymentRequest(array $bodyValues)
+    {
+        $transactionInfo = $this->commandBus->handle(new GetApplePayPaymentRequestQuery(new CartId($this->context->cart->id)));
```
Comment (seiwan) [inline controllers/front/applepay.php:70]: Same, idiomatic code ?

File: src/PayPal/ApplePay/Query/GetApplePayPaymentRequestQueryHandler.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * @param GetApplePayPaymentRequestQuery $query
+     *
+     * @return GetApplePayPaymentRequestQueryResult
+     *
+     * @throws PsCheckoutException
+     */
+    public function handle(GetApplePayPaymentRequestQuery $query)
+    {
+        $cartPresenter = (new CartPresenter())->present();
```
Comment (seiwan) [inline src/PayPal/ApplePay/Query/GetApplePayPaymentRequestQueryHandler.php:49]: Idiomatic code ?

File: src/PayPal/ApplePay/Query/GetApplePayPaymentRequestQueryHandler.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        $orderPayloadBuilder = new OrderPayloadBuilder($cartPresenter);
+
+        $orderPayloadBuilder->buildFullPayload();
+        $payload = $orderPayloadBuilder->presentPayload();
+
+        return new GetApplePayPaymentRequestQueryResult($this->builder->buildMinimalPaymentRequestFromPayPalPayload($payload->getArray()));
```
Comment (seiwan) [inline src/PayPal/ApplePay/Query/GetApplePayPaymentRequestQueryHandler.php:55]: Idiomatic code ?

**Review by Matt75 — APPROVED** [review]

**Pull request merged**